ivadomed/prepare_dataset.py

```diff
@@ -0,0 +1,215 @@
+    :return: list of images
+    """
+    if type(img_list) != list:
+        img_list = [img_list]
+    img_zero_padding_list = []
+    for i in range(len(img_list)):
+        img = img_list[i]
+        img_tmp = np.zeros((x_val, y_val, 1), dtype=np.float64)
+        img_tmp[0:img.shape[0], 0:img.shape[1], 0] = img
+        img_zero_padding_list.append(img_tmp)
+
+    return img_zero_padding_list
+
+
+def mask2label(path_label, aim='full'):
+    """
+    Convert nifti image to an array of coordinates
+    :param path_label: path of nifti image
+    :return:
+    """
+    image = nib.load(path_label)
+    image = nib.as_closest_canonical(image)
+    arr = np.array(image.dataobj)
+    list_label_image = []
+    for i in range(len(arr.nonzero()[0])):
+        x = arr.nonzero()[0][i]
+        y = arr.nonzero()[1][i]
+        z = arr.nonzero()[2][i]
+        if aim == 'full':
+            if arr[x, y, z] <30 and arr[x, y, z] != 1:
+                list_label_image.append([x, y, z, arr[x, y, z]])
+        elif aim == 'c2':
+            if arr[x, y, z] == 3:
+                list_label_image.append([x, y, z, arr[x, y, z]])
+    list_label_image.sort(key=lambda x : x[3])
+    return list_label_image
+
+
+def get_midnifti(path_im, ind):
+    """
+    Retrieve the input images for the network. This images are generated by
+    averaging the 7 slices in the middle of the volume
+    :param path_im: path to image
+    :param ind: index of the middle
+    :return:
+    """
+    image = nib.load(path_im)
+    image = nib.as_closest_canonical(image)
+    arr = np.array(image.dataobj)
+    return np.mean(arr[ind - 3:ind + 3, :, :], 0)
+
+
+def images_normalization(img_list, std=True):
+    if type(img_list) != list:
+        img_list = [img_list]
+    img_norm_list = []
+    for i in range(len(img_list)):
+        # print('Normalizing ' + str(i + 1) + '/' + str(len(img_list)))
+        img = img_list[i] - np.mean(img_list[i])  # zero-center
+        if std:
+            img_std = np.std(img)  # normalize
+            epsilon = 1e-100
+            img = img / (img_std + epsilon)  # epsilon is used in order to avoid by zero division
+        img_norm_list.append(img)
+    return img_norm_list
+
+
+def extract_all(list_coord_label, shape_im=(1, 150, 200)):
+    """
+    Create groundtruth by creating gaussian Function for every ground truth points for a single image
+    :param list_coord_label: list of ground truth coordinates
+    :param shape_im: shape of output image with zero padding
+    :return: a 2d heatmap image.
+    """
+    shape_tmp = (1, shape_im[0], shape_im[1])
+    final = np.zeros(shape_tmp)
+    for x in list_coord_label:
+        train_lbs_tmp_mask = label2maskmap_gt(x, shape_tmp)
+        for w in range(shape_im[0]):
+            for h in range(shape_im[1]):
+                final[0, w, h] = max(final[0, w, h], train_lbs_tmp_mask[w, h])
+    return final
+
+
+def transform_dataset(bids_path, suffix, aim):
```

Comment:
need to be more specific
```suggestion
def extract_mid_slice_and_convert_coordinates_to_heatmaps(bids_path, suffix, aim):
```